plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -430,6 +439,16 @@ public IcebergMetadata(
         this.metastoreFactory = requireNonNull(metastoreFactory, "metastoreFactory is null");
         this.addFilesProcedureEnabled = addFilesProcedureEnabled;
         this.allowedExtraProperties = requireNonNull(allowedExtraProperties, "allowedExtraProperties is null");
+
+        if (metadataParallelism == 1) {
+            this.executorService = directExecutor();
+        }
+        else {
+            this.executorService = Executors.newFixedThreadPool(metadataParallelism, new ThreadFactoryBuilder()
+                    .setNameFormat("iceberg-metadata-pool-%d")
+                    .setDaemon(true) // Optional: Makes threads daemon threads.
+                    .build());
```

Comment:
Why do we initialize `Executor` in `IcebergMetadata` instead of providing it in `IcebergModule`?

Reply:
I have just moved that to `IcebergMetadataFactory` and use the already existing `createIcebergMetadataExecutor` that is located in `IcebergModule`